packages/openapi3/src/schema-emitter.ts

```diff
@@ -451,33 +447,12 @@ export class OpenAPI3SchemaEmitter extends TypeEmitter<
 
   enumDeclaration(en: Enum, name: string): EmitterOutput<object> {
     const baseName = getOpenAPITypeName(this.emitter.getProgram(), en, this.#typeNameOptions());
-    return this.#createDeclaration(en, baseName, new ObjectBuilder(this.#enumSchema(en)));
+    return this.#createDeclaration(en, baseName, new ObjectBuilder(this.enumSchema(en)));
   }
 
-  #enumSchema(en: Enum): OpenAPI3Schema {
-    const program = this.emitter.getProgram();
-    if (en.members.size === 0) {
-      reportDiagnostic(program, { code: "empty-enum", target: en });
-      return {};
-    }
-
-    const enumTypes = new Set<JsonType>();
-    const enumValues = new Set<string | number>();
-    for (const member of en.members.values()) {
-      enumTypes.add(typeof member.value === "number" ? "number" : "string");
-      enumValues.add(member.value ?? member.name);
-    }
-
-    if (enumTypes.size > 1) {
-      reportDiagnostic(program, { code: "enum-unique-type", target: en });
-    }
-
-    const schema: OpenAPI3Schema = {
-      type: enumTypes.values().next().value!,
-      enum: [...enumValues],
-    };
-
-    return this.#applyConstraints(en, schema);
+  enumSchema(en: Enum): Schema {
```

Comment:
Ah - I intentionally didn't add this one in base because this function will actually be different between 3.1 and 3.0 once we support allowing `types` as an array.  (https://github.com/microsoft/typespec/issues/5011 specifically)

Reply:
Got it